lib/semantics/check-arithmeticif.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "attr.h"
+#include "check-arithmeticif.h"
+#include "scope.h"
+#include "semantics.h"
+#include "symbol.h"
+#include "tools.h"
+#include "type.h"
+#include "../evaluate/traversal.h"
+#include "../parser/message.h"
+#include "../parser/parse-tree.h"
+
+namespace Fortran::semantics {
+
+class ArithmeticIfStmtContext {
+public:
+  ArithmeticIfStmtContext(SemanticsContext &context)
+    : messages_{context.messages()} {}
+
+  bool operator==(const ArithmeticIfStmtContext &x) const { return this == &x; }
+  bool operator!=(const ArithmeticIfStmtContext &x) const { return this != &x; }
+
+  // Arithmetic IF statements have been removed from Fortran 2018.
+  // The constraints and requirements here refer to the 2008 spec.
```

Comment:
Maybe add some comments next to each constraint that is checked for traceability.  (It will have any future efforts to audit the code.)

Reply:
I put the constraint and requirement references before the first if.